src/probot.ts

```diff
@@ -74,6 +74,7 @@ export class Probot {
     const octokit = new Octokit({
       request: options.request,
     });
+    octokit.log = this.log.child({ name: "octokit" });
```

Comment:
Use the `log` constructor argument instead: https://github.com/octokit/core.js/#logging

```suggestion
    const octokit = new Octokit({
      request: options.request,
      log: this.log.child({ name: "octokit" }),
    });
```

I'm not sure if this works by now, I think it doesn't due to pino's method binding, we might need to do this instead

```suggestion
    const octokitLogger = this.log.child({ name: "octokit" }),
    const octokit = new Octokit({
      request: options.request,
      log: {
        debug: octokitLogger.debug.bind(octokitLogger),
        info: octokitLogger.info.bind(octokitLogger),
        warn: octokitLogger.warn.bind(octokitLogger),
        error: octokitLogger.error.bind(octokitLogger),
      }
    });
```

Setting `octokit.log` directly to a pino instance is tempting, but if we accidentally start using pino-only methods on octokit.log we could end up with hard-to-debug errors.

Reply:
I had that before, and it broke. But I'll have another look here considering it's still broken, just less noticeable.